library/bigip_ssl_certificate.py

```diff
@@ -0,0 +1,322 @@
+
+
+class BigIPRest(BigIPCommon):
+    def connection(self, hostname, username, password, verify):
+        return ManagementRoot(hostname=hostname,
+                              username=username,
+                              password=password)
+
+    def delete(self, name):
+        result = False
+        r = self.delete_cert(name + '.crt')
+        if r:
+            result = True
+        r = self.delete_key(name + '.key')
+        if r:
+            result = True
+        return result
+
+    def delete_cert(self, name):
+        try:
+            c = self.api.tm.sys.crypto.certs.cert.load(name=name)
+            c.delete()
+        except iControlUnexpectedHTTPError as e:
+            if e.response.status_code == 404:
+                return False
+            else:
+                raise F5ModuleError(e)
+        else:
+            return True
+
+    def delete_key(self, name):
+        try:
+            k = self.api.tm.sys.crypto.keys.key.load(name=name)
+            k.delete()
+        except iControlUnexpectedHTTPError as e:
+            if e.response.status_code == 404:
+                return False
+            else:
+                raise F5ModuleError(e)
+        else:
+            return True
+
+    def exists_cert(self, name):
+        return self.api.tm.sys.crypto.certs.cert.exists(name=name)
+
+    def exists_key(self, name):
+        return self.api.tm.sys.crypto.keys.key.exists(name=name)
+
+    def import_cert(self, **kwargs):
+        name = kwargs['name']
+        cert_format = kwargs['cert_format']
+        if cert_format == 'pem':
+            try:
+                certfilename = kwargs['certfilename']
+            except KeyError:
+                certfilename = kwargs['name'] + '.crt'
+            else:
+                self.upload(certfilename)
+                self.install_cert(name, os.path.basename(certfilename))
+            try:
+                keyfilename = kwargs['keyfilename']
+            except KeyError:
+                pass
+            else:
+                self.upload(keyfilename)
+                self.install_key(name, os.path.basename(keyfilename))
+            return True
+        elif cert_format == 'pkcs12':
+            raise F5ModuleError("pkcs12 not supported by f5-sdk")
+
+    def install_cert(self, name, certfilename):
+        filename = os.path.join('/var/config/rest/downloads', certfilename)
+        params = {'from-local-file': filename, 'name': name}
+        self.api.tm.sys.crypto.certs.exec_cmd('install', **params)
+
+    def install_key(self, name, keyfilename):
+        filename = os.path.join('/var/config/rest/downloads', keyfilename)
+        params = {'from-local-file': filename, 'name': name}
+        self.api.tm.sys.crypto.keys.exec_cmd('install', **params)
+
+    def upload(self, filename):
+        self.api.shared.file_transfer.uploads.upload_file(filename)
+
+
+def main():
+    module = AnsibleModule(
```

Comment:
With the inclusion of `from ansible.module_utils.f5 import *`, this argument spec can be simplified.

If you take a look here

https://github.com/F5Networks/f5-ansible/blob/master/library/bigip_device_sshd.py#L335

from line 335 to line 360, you can use the same workflow to minimize this block.